opm/autodiff/WellHelpers.hpp

```diff
@@ -191,6 +191,63 @@ namespace Opm {
             return retval;
         }
 
+        template <class MatrixType>
+        bool invertMatrixLapack(MatrixType& matrix) {
+
+            const int m = matrix.M();
+            const int n = matrix.N();
+            assert(m == n); // it needs to be a square matrix
+
+            // the matrix
+            double* A = new double[m * n];
+
+            // copying the data in matrix to A
+            int i_count = 0;
+            for (int irow = 0; irow < m; ++irow) {
+                for (int icol = 0; icol < n; ++icol) {
+                    A[i_count] = matrix[irow][icol];
+                    ++i_count;
+                }
+            }
+
+            int* ipiv = new int[n + 1];
+            int lwork = m * n;
+            double* work = new double[lwork];
+            int info;
```

Comment:
Not a fan of explicit `new` and (especially) `delete`.  Is there a reason for not using `std::vector<>` or, in a pinch, `std::unique_ptr<T[]>` here?

Reply:
This is very good suggestion. It was a reaction when seeing C function interface.